lib/index.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
+const RateLimiter = require('./rate-limiter');
+
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    return RateLimiter.limit(request, options.redisClient, options.defaultRate)
+    .then((rate) => {
+      if (rate && rate.current > rate.limit) {
+        return reply(options.overLimitHandler());
+      }
 
-    return reply.continue();
+      request.payload.rate = rate;
```

Comment:
just for testing purposes to see if the rate limiting actually went as expected - in the next PR we'd be setting the headers